tools/flexynesis/flexynesis.xml

```diff
@@ -0,0 +1,596 @@
+                $data_names.append("omics_" + str($i))
+            ## #end if
+        #end for
+        flexynesis  
+                    --data_path 'input'
+                    --outdir 'output'
+                    --model_class $model_class
+                    #if str($model_class) == 'GNN':
+                        --gnn_conv_type $gnn_conv_type
+                        --string_organism $string_organism
+                        --string_node_name $string_node_name
+                    #end if
+                    #if str($training_type.model) == 's_train':
+                        #if str($target_variables) != '':
+                            --target_variables $target_variables
+                        #end if
+                        #if str($surv_event_var) != '':
+                            --surv_event_var $surv_event_var
+                            --surv_time_var $surv_time_var
+                        #end if
+                    #end if
+                    #if str($training_type.model) == 'cm_train':
+                        --input_layers $input_layers
+                        --output_layers $output_layers
+                    #end if
+                    --fusion_type $fusion_type
+                    --hpo_iter $hpo_iter
+                    --finetuning_samples $finetuning_samples
+                    --variance_threshold $variance_threshold
+                    --correlation_threshold $correlation_threshold
+                    --subsample $subsample
+                    --features_min $features_min
+                    --features_top_percentile $features_top_percentile
+                    --data_types $str(",".join($data_names))
+                    --early_stop_patience $early_stop_patience
+                    --hpo_patience $hpo_patience
+                    $log_transform
+                    $use_loss_weighting
+                    $use_cv
+                    $evaluate_baseline_performance
+                    $disable_marker_finding
+                    ## "\${GALAXY_FLEXYNESIS_EXTRA_ARGUMENTS:-''}"
+    ]]></command>
+    <inputs>
+        <param name="non_commercial_use" label="I certify that I am not using this tool for commercial purposes." type="boolean" truevalue="NON_COMMERCIAL_USE" falsevalue="COMMERCIAL_USE" checked="False">
+            <validator type="expression" message="This tool is only available for non-commercial use.">value == True</validator>
+        </param>
+        <param name="train_clin" type="data" format="csv" label="Training clinical data"/>
+        <param name="test_clin" type="data" format="csv" label="Test clinical data"/>
+        <param name="train_omics_main" type="data" format="csv" label="Training omics data"/>
+        <param name="test_omics_main" type="data" format="csv" label="Test omics data"/>
+        <param name="assay_main" type="text" optional="true" label="What type of assay is your input?" help="This would be used as output name."/>
```

Comment:
You might want to prevent whitespaces here, just letters?

Reply:
Yes, I'll add that